Config/Filter/Type/Thumbnail.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\Config\Filter\Type;
+
+use Liip\ImagineBundle\Config\FilterInterface;
+
+final class Thumbnail implements FilterInterface
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var array
+     */
+    private $size;
+
+    /**
+     * @var string
+     */
+    private $mode;
+
+    /**
+     * @var bool
+     */
+    private $allowUpscale;
+
+    /**
+     * @var string
+     */
+    private $filter;
+
+    /**
+     * @param string      $name
+     * @param array       $size
```

Comment:
can you specify what kind of array this is? is it a hashmap with width and height?